src/ts/app/storage/src/state-transition.ts

```diff
@@ -0,0 +1,59 @@
+import {FileChangeType, Subtree} from "./subtree"
+import produce from "immer"
+
+export function initSubtree(): Subtree {
+    return {
+        documentCollection: {
+            documents: [],
+            nameToDocument: {}
+        },
+
+        lastDocumentChangeIndexApplied: 0,
+        documentChanges: []
+    }
+}
+
+
+export function fileDropped(previous: Subtree, name: string, content: string): Subtree {
+    return applyNextFileChanges(
+        produce(previous, draft => {
+            draft.documentChanges.push({
+                type: FileChangeType.ADD_PYTHON_MODULE,
+                name: name,
+                content: content
+            })
+        }))
+}
+
+export function applyNextFileChanges(previous: Subtree) {
+    return produce(previous, draft => {
+        previous.documentChanges.slice(previous.lastDocumentChangeIndexApplied).forEach((fileChange) => {
+            switch (fileChange.type) {
+                case(FileChangeType.ADD_PYTHON_MODULE):
```

Comment:
just want to make sure, this doesn't colide with the versioning mechanism you plan right? currently we have bugs because the filename has the version in it, and so if someone refreshes the page to an older version or something you start getting a bunch of files with different versions on the end because the file contents don't match.

Ideally, version is in a different place, and if there is a conflict, users select one or the other to work on...